### PR TITLE
Move pkg/api/unversioned to pkg/apis/meta/v1

### PR DESCRIPTION
**Pull request by smarterclayton**
This moves code from using pkg/api/unversioned to pkg/apis/meta/v1 with the `metav1` local package name.

Built on top of #37532 (the first three commits related to ExportOptions)

Part of #37530

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37534)

### CODE REVIEW

File: cmd/libs/go2idl/client-gen/test_apis/testgroup/types.generated.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"fmt"
 	codec1978 "github.com/ugorji/go/codec"
 	pkg2_api "k8s.io/kubernetes/pkg/api"
-	pkg1_unversioned "k8s.io/kubernetes/pkg/api/unversioned"
+	pkg1_unversioned "k8s.io/kubernetes/pkg/apis/meta/v1"
```
Comment (sttts) [inline cmd/libs/go2idl/client-gen/test_apis/testgroup/types.generated.go]: s/unversioned/metav1/

**Comment by caesarxuchao** [comment]
lgtm. cc @lavalamp if you have comments on the group name, not sure if you were OOO when the issue was discussed.

**Comment by smarterclayton** [comment]
Note that if we are comfortable (but not 100% sure) about the group name, this PR does not expose it publicly so we can still change it later.

**Comment by smarterclayton** [comment]
@k8s-bot node e2e test this

**Comment by smarterclayton** [comment]
Green and up to date - applying label

**Comment by smarterclayton** [comment]
@k8s-bot cvm gce e2e test this

Port forward flake.

**Pull request merged**